coal_petroleum/demo/sale_order_post.xml

```diff
@@ -0,0 +1,90 @@
+            obj().search([('state', '=', 'assigned'), ('id', 'in',
+                (
+                    obj().env.ref('coal_petroleum.sale_order_1') +
+                    obj().env.ref('coal_petroleum.sale_order_4')
+                ).procurement_group_id.stock_move_ids.picking_id.ids)
+            ]).ids"/>
+    </function>
+
+    <function name="create" model="x_quality_check_worksheet_template_1_studio">
+        <value model="x_quality_check_worksheet_template_1_studio" eval="dict(
+            obj().default_get(list(obj().fields_get())),
+            **{
+                'x_quality_check_id': obj().env.ref('coal_petroleum.sale_order_1').picking_ids[1].check_ids.id,
+                'x_studio_test_type_1': 'ARB - As-Received Basis',
+                'x_studio_net_calorific_value': 11.0,
+                'x_studio_ash_content': 11.0,
+                'x_studio_moisture_content': 11.0,
+                'x_studio_test_type': 11.0,
+                'x_studio_total_sulphur': 11.0,
+                'x_studio_flying_substance': 11.0,
+                'x_studio_hardgrave_grindability_index': 11.0,
+            }
+        )"/>
+    </function>
+    <function name="create" model="x_quality_check_worksheet_template_1_studio">
+        <value model="x_quality_check_worksheet_template_1_studio" eval="dict(
+            obj().default_get(list(obj().fields_get())),
+            **{
+                'x_quality_check_id': obj().env.ref('coal_petroleum.sale_order_4').picking_ids[1].check_ids.id,
+                'x_studio_test_type_1': 'ARB - As-Received Basis',
+                'x_studio_net_calorific_value': 20.0,
+                'x_studio_ash_content': 20.0,
+                'x_studio_moisture_content': 20.0,
+                'x_studio_test_type': 20.0,
+                'x_studio_total_sulphur': 20.0,
+                'x_studio_flying_substance': 20.0,
+                'x_studio_hardgrave_grindability_index': 20.0,
+            }
+        )"/>
+    </function>
+
+    <function name="do_pass" model="quality.check">
+        <value model="quality.check" eval="
+            obj().search([('picking_id', 'in',(
+                obj().env.ref('coal_petroleum.sale_order_1') +
+                obj().env.ref('coal_petroleum.sale_order_4')
+            ).picking_ids.ids)]).ids"/>
+    </function>
+
+    <function name="button_validate" model="stock.picking">
+        <value model="stock.picking" eval="
+            obj().search([('state', '=', 'assigned'), ('id', 'in',
+                (
+                    obj().env.ref('coal_petroleum.sale_order_1') +
+                    obj().env.ref('coal_petroleum.sale_order_4')
+                ).picking_ids.ids)
+            ]).ids"/>
+    </function>
+
+    <function name="_create_invoices" model="sale.order">
+        <value eval="[ref('sale_order_1')]"/>
+    </function>
+
+    <function name="_create_invoices" model="sale.order">
+        <value eval="[ref('sale_order_4')]"/>
```

Comment:
@vava-odoo 

No, we shouldn't perform grouping in this scenario. 

In both Sales Orders, the customer is the same. If we group them and invoke the _create_invoices function, it will generate a single invoice incorporating all the products from both Sales Orders. 

If you have any suggestions, please let me know.